RudderStack for Currents - MERGE 3/7 (#4933)
braze-inc/braze-docs: _docs/_partners/data_and_infrastructure_agility/customer_data_platform/rudderstack/rudderstack.md

| --- | --- |
| RudderStack account | A [RudderStack account](https://app.rudderstack.com/) is required to take advantage of this partnership. |
| Configured source | A [source][3] is essentially the origin of any data sent to RudderStack, such as websites, mobile apps, or backend servers. You are required to configure the source before setting up Braze as a destination in RudderStack. |
| Braze REST API key | A Braze REST API key with `users.track`, `users.identify`, and `users.alias.new` permissions.<br><br>This can be created within the **Braze Dashboard > Developer Console > REST API Key > Create New API Key**. |
| Braze app key | To get your app key, navigate to **Braze Dashboard > Developer Console > Identification** and find your app name. Save the associated identifier string.
| Data center | Your data center aligns with your Braze dashboard [instance][15].  |
{: .reset-td-br-1 .reset-td-br-2 .reset-td-br-3  .reset-td-br-4}

## Integration

### Step 1: Add a source

To start sending data to Braze, you first need to make sure a source is set up in your RudderStack app. Visit [RudderStack][22] to learn how to set up your data source.

### Step 2: Configure destination

> Now that you have your data source set up, in the RudderStack dashboard, select **ADD DESTINATION** under **Destinations**. From the list of available destinations, select **Braze**and click **Next**.

Reviewer: ```suggestion
Now that you have your data source set up, in the RudderStack dashboard, select **ADD DESTINATION** under **Destinations**. From the list of available destinations, select **Braze** and click **Next**.
```

In the Braze destination, provide the app key, Braze REST API key, data cluster, and native SDK option (device mode only). The native SDK option will use the Braze native SDK to send events if toggled on. 

![][0]{: style="max-width:40%;margin-bottom:15px;"}

## Step 3: Choose the type of integration

You can choose to integrate RudderStack's web and native client-side libraries with Braze using either a side-by-side (device mode) integration or a server-to-server (cloud mode) integration.

- Integration type

>   - [Side-by-Side / Device Mode](#device-mode): RudderStack will send the event data to Braze directly from your client (browser or mobile application).

Author: ```suggestion
  - [Side-by-side / device mode](#device-mode): RudderStack will send the event data to Braze directly from your client (browser or mobile application).
```

  - [Server-to-Server / Cloud Mode](#cloud-mode): The Braze SDK sends the event data directly to RudderStack, which is then transformed and routed to Braze.

{% alert note %} 

> Learn more about RudderStack's [connection modes](https://rudderstack.com/docs/connections/rudderstack-connection-modes/) and the benefits of each.

Reviewer: This link is broken

> Learn more about RudderStack's [connection modes](https://rudderstack.com/docs/connections/rudderstack-connection-modes/) and the benefits of each.

Author: ```suggestion
Learn more about RudderStack's [connection modes](https://www.rudderstack.com/docs/destinations/rudderstack-connection-modes/) and the benefits of each.
```

{% endalert %}

### Step 3a: Side-by-side integration (device mode) {#device-mode}

With this mode, you can send your events to Braze using the Braze SDK set up on your website or mobile app.

Set up the mappings to the RudderStack SDK for [Android](https://github.com/rudderlabs/rudder-integration-braze-android), [iOS](https://github.com/rudderlabs/rudder-integration-braze-ios), or [React Native] on Braze's GitHub repository, as described in step 4. 

To complete the device mode integration, refer to the detailed RudderStack instructions for [adding Braze to your project](https://rudderstack.com/docs/destinations/marketing/braze/#adding-device-mode-integration).

### Step 3b: Server-to-server integration (cloud mode) {#cloud-mode}

In this mode, the SDK sends the event data directly to the RudderStack server. RudderStack then transforms this data and routes it to the desired destination. This transformation is done in the RudderStack backend, using RudderStack's Transformer module.

To enable the integration, you will need to map the RudderStack methods to Braze, as described in step 4.

{% alert note %} 

> RudderStack's server-side SDKs (Java, Python, Node.js, Go, Ruby) support only Cloud Mode. This is because their server-side SDKs operate in the RudderStack backend and cannot load any Braze-specific SDK. 

Reviewer: Some minor capitalization inconsistencies here, we have "Cloud Mode" and "cloud mode". Not sure which is correct but I'm suggesting lowercase here since that's what you have for device mode 👀 

```suggestion
RudderStack's server-side SDKs (Java, Python, Node.js, Go, Ruby) support only cloud mode. This is because their server-side SDKs operate in the RudderStack backend and cannot load any Braze-specific SDK. 
```
